@here/olp-sdk-dataservice-read/lib/client/PartitionsRequest.ts

```diff
@@ -23,74 +23,102 @@ import { validateBillingTag, validatePartitionsIdsList } from "..";
+            */
+           public withBillingTag(tag: string): PartitionsRequest {
+               this.billingTag = validateBillingTag(tag);
+               return this;
+           }
 
-    /**
-     * Gets a billing tag to group billing records together.
-     *
-     * @return The `BillingTag` string.
-     */
-    public getBillingTag(): string | undefined {
-        return this.billingTag;
-    }
+           /**
+            * Gets a billing tag to group billing records together.
+            *
+            * @return The `BillingTag` string.
+            */
+           public getBillingTag(): string | undefined {
+               return this.billingTag;
+           }
 
-    /**
-     * A setter for the provided partition IDs.
-     *
-     * @param ids The ID of partitions from which you want to get metadata.
-     *
-     * The required quantity is between 1 and 100.
-     * If partition IDs are not provided, all partitions are retrieved.
-     * If the quantity of partition IDs is less than 1 (empty array) or more than 100, an error is thrown.
-     * @returns The updated [[PartitionsRequest]] instance that you can use to chain methods.
-     */
-    public withPartitionIds(ids: string[]): PartitionsRequest {
-        this.partitionIds = validatePartitionsIdsList(ids);
-        return this;
-    }
+           /**
+            * A setter for the provided partition IDs.
+            *
+            * @param ids The ID of partitions from which you want to get metadata.
+            *
+            * The required quantity is between 1 and 100.
+            * If partition IDs are not provided, all partitions are retrieved.
+            * If the quantity of partition IDs is less than 1 (empty array) or more than 100, an error is thrown.
+            * @returns The updated [[PartitionsRequest]] instance that you can use to chain methods.
+            */
+           public withPartitionIds(ids: string[]): PartitionsRequest {
+               this.partitionIds = validatePartitionsIdsList(ids);
+               return this;
+           }
 
-    /**
-     * Gets partition IDs for the request.
-     *
-     * @return The `partitionIds` string.
-     */
-    public getPartitionIds(): string[] | undefined {
-        return this.partitionIds;
-    }
-}
+           /**
+            * Gets partition IDs for the request.
+            *
+            * @return The `partitionIds` string.
+            */
+           public getPartitionIds(): string[] | undefined {
+               return this.partitionIds;
+           }
+           
+           /**
+            * A setter for the provided additional fields.
+            * 
+            * @param dataSize 
+            * @param checksum 
+            * @param compressedDataSize 
+            * 
+            * @returns The updated [[PartitionsRequest]] instance that you can use to chain methods.
+            */
+           public withAdditionalFields(
+               dataSize: string,
```

Comment:
Please use array instead of vars, Array<'dataSize' | 'checksum' | 'compressedDataSize' | 'crc'>

Reply:
Added